samza-azure/src/main/java/org/apache/samza/system/azureblob/AzureBlobConfig.java

```diff
@@ -80,6 +80,12 @@ public class AzureBlobConfig extends MapConfig {
   public static final String SYSTEM_MAX_FLUSH_THRESHOLD_SIZE = SYSTEM_AZUREBLOB_PREFIX + "maxFlushThresholdSize";
   private static final int SYSTEM_MAX_FLUSH_THRESHOLD_SIZE_DEFAULT = 10485760;
 
+  // initialization size of in-memory OutputStream
+  // This value should be between SYSTEM_INIT_BUFFER_SIZE_DEFAULT and getMaxFlushThresholdSize() exclusive.
+  public static final String SYSTEM_INIT_BUFFER_SIZE = SYSTEM_AZUREBLOB_PREFIX + "initBufferSize.bytes";
+  // re-use size for parameterless constructor java.io.ByteArrayOutputStream()
+  public static final int SYSTEM_INIT_BUFFER_SIZE_DEFAULT = 32;
```

Comment:
Can you provide some context through comments here on why and how this value is chosen and what its impact is?
E.g., what was the previous default provided through config or defaulted somehow (through code) that caused issues so that folks can be mindful about making changes to these defaults in future.

Reply:
@mynameborat Can you provide an example? The comment above the field identifies the source. tbc, `ByteArrayOutputStream` has two constructors, and the value here is the default when using the [parameterless constructor](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/ByteArrayOutputStream.html#%3Cinit%3E()). 

Fwiw, I am in-part also avoiding "magic" values. Where the implementation, `AzureBlobOutputStream`, would need to "understand" if the value is provided/set, ie. `-1` or `null`, and then select the "correct" constructor. This allows the config object to establish the "meaning"/intent of the value and create a "default" that is always usable by the implementation.

Reply:
There is a [1] configuration-table.html in the code base where you can document something about the config you introduced above and also explain about the details.

[1] https://github.com/apache/samza/blob/master/docs/learn/documentation/versioned/jobs/configuration-table.html

Reply:
@mynameborat ty for pointing to the file.

Reply:
@mynameborat I didn't see any examples of GC warnings or notes. I added instructions for when the value should be changed and what else (outside of samza) might need to be changed too.

Reply:
Sounds good.